### DISCUSSION TITLE
Invalid state token, when login in to concourse.

### DISCUSSION THREAD

**Discussion by NoobDela** [concourse/concourse#5871] [Help & Support]
Hi guys, I’m following the normal tutorial for concourse, it was working fine until a few days ago. When logging into concourse it brings me to a white page saying “invalid state token”. I’ve cleared cache cookies, and I even destroyed the project and created a new one and that hasn’t worked. I've deleted the target from ~/.flyrc and recreated. I tried other browsers and in incognito. I can't seem to find a solution. from the looks of it the authentication is working, but my browser(Chrome, Firefox) isn't setting the state token maybe, any ideas? Any help would be appreciated. Thank you!

Edit: Edit: Finally figured it out. It's a bug. Because I start docker-compose with the external url as http://localhost:8080, but then I set my target login as http://127.0.0.1:8080 and log in with http://127.0.0.1:8080 it causes the sort of error. It has to match the external url, so I changed it and it works.

**Comment by haydonryan** [comment]
I ran into this.

```
 fly -t ci login 
```
gave unexpected token.

```
fly -t ci login -c=https://ci.URL.com
```
This worked, definitely a bug, and still here 5 years later.

**Reply by taylorsilva** [reply]
Interesting, so it sounds like Concourse gave a token for a URL that didn't match the external URL. I guess the bug here is that Concourse should not have issued the token in the first place unless it matched the external URL it is configured with.

**Reply by haydonryan** [reply]
yah - i even tried to removed the contents of .flyrc too, wondering if it was that.  It's a minor nibble though.
